util/stream.go

```diff
@@ -156,40 +136,76 @@ func (m *MessageStream) inbound() {
 			return
 		}
 
-		for i := 0; i < n; i++ {
-			if hdr < 4 {
-				hdrBuf[hdr] = tmpBuf[i]
-				buf.WriteByte(tmpBuf[i])
-				hdr += 1
-				if hdr >= 4 {
-					// MessageStream is not protocol agnostic. Reading length based
-					// on OpenFlow header field.
-					msgLen = int(binary.BigEndian.Uint16(hdrBuf[2:])) - 4
+		// Append the bytes read from the connection to buf.
+		buf.Write(tmpBuf[:n])
+
+		// Read from the connection until the OpenFlow message header is retrieved.
+		for buf.Len() >= 4 {
```

Comment:
I think the code would be much more readable if you had:
```go
if buf.Len() < 4 {
        continue
}
```

it actually took me a while to understand the current version